x-pack/plugins/uptime/server/lib/adapters/monitors/elasticsearch_monitors_adapter.ts

```diff
@@ -187,35 +180,37 @@ export class ElasticsearchMonitorsAdapter implements UMMonitorsAdapter {
       },
     };
 
-    const queryResult = await this.database.search(request, params);
-    const hostBuckets = get(queryResult, 'aggregations.urls.buckets', []);
-    const monitorStatuses = hostBuckets.map(bucket => {
-      const latest = get(bucket, 'latest.hits.hits', []);
-      return latest.reduce(
-        (acc, doc) => {
-          const status = get(doc, '_source.monitor.status', null);
-          if (statusFilter && statusFilter !== status) {
-            return acc;
-          }
+    let up: number = 0;
+    let down: number = 0;
+    let searchAfter: any = null;
+
+    do {
+      if (searchAfter) {
+        set(params, 'body.aggs.ids.composite.after', searchAfter);
+      }
+
+      const queryResult = await this.database.search(request, params);
+      const idBuckets = get(queryResult, 'aggregations.ids.buckets', []);
+
+      if (idBuckets.length === 0) {
+        break;
+      }
+
+      idBuckets.forEach(bucket => {
+        // We only get the latest doc
+        const status = get(bucket, 'latest.hits.hits[0]._source.monitor.status', null);
+        if (statusFilter && statusFilter !== status) {
```

Comment:
```suggestion
        if (!statusFilter || (statusFilter && statusFilter === status)) {
```
We need to allow the code to run when no filter is present. Additionally, the filter condition is incorrect so we are returning `up` counts when filtering for `down` and visa versa.